syntax = "proto3";

package TW.LTO.Proto;
option java_package = "wallet.core.jni.proto";

// Input data necessary to create a signed transaction.
message SigningInput {
    int64 amount = 1;

    int64 fee = 2;

    string to = 3;

    // any 140 bytes payload, will be displayed to the client as utf-8 string
    bytes attachment = 4;

    int64 timestamp = 5;

    bytes public_key = 6;

    bytes private_key = 7;
}

// Transaction signing output.
message SigningOutput {
    bytes signature = 1;
}
